AirSkyBoat/AirSkyBoat
# Adds besieged mob advance stage

**_I affirm:_**


## Please enter a player-facing description

Adds besieged mob advance stage

What works:

- Mobs spawn in the corresponding zone when they reach advance stage
- Only one beastmen stronghold can advance at once
- Mobs path to alzhabi, and when reaching alzhabi move to "attacking" stage
- If mobs are all defeated, stronhold mobs to "retreat" stage

What needs work:

- There is only one mob path per stronghold, and it is not polished
- Mobs all spawn at the same time. Should be staggered.
- Mobs should not continue fighting a player when engaged. They should lose interest after a few hits and return to their path.
- The kinds of mob spawned should be chosen more carefully (currently just going down the list)

## Steps to test these changes

- Enable Besieged and besieged logging via settings
- Wait for a force level to reach 100
- Observe beastmen advance message and go to the zone
- Look at logs and go to any of the spawned mob ids
- Either:
   - kill all mobs, or use `!besieged retreat` and watch the retreat message
   - OR 
   - Let mobs slowly path to alzhaby. Call `!besieged speed 255` to speed it up
